### ISSUE TITLE
Create style `font` that all other font properties inherit from.

### ISSUE THREAD

**Issue by domehead100** [TonyGermaneri/canvas-datagrid#157]
### Expected behavior and actual behavior.
For one thing, I'd like to be able to set the various types of cell fonts using a fully compatible CSS font property syntax, which includes the font weight, etc.  If I put a font weight in, everything falls back to the default font (10px sans-serif).

Secondly, is there a way to just specify once the default font for the entire grid, instead of having to cover all of the various corner cases?

### Steps to reproduce the problem.
Try and set any of the cell fonts, such as cellFont, to a valid CSS font property string that contains a font weight, and see that the grid does not honor the font.  For example:

```"bold 14px Roboto, \"Trebuchet MS\", Arial, Helvetica, sans-serif"```

### Specifications like the version of the project, operating system, or hardware.

Latest version, Windows 10.

### Discussion
Looks like you're parsing the font strings yourself, and you're not expecting, for example, a font weight to be passed in.  

I assume that you're parsing the font to extract the font height so that you can adjust things based on the scale property:
```css
self.ctx.font = (self.style[cellStyle + 'FontHeight'] * self.scale) + 'px ' + self.style[cellStyle + 'FontName'];
```
Also, you're assuming that all font sizes are going to be in "px".  I'd recommend taking a look at an external CSS font property parser, such as https://github.com/jedmao/parse-css-font.  I'm not sure how you could implement the scale feature with font sizes that are not in "px", but I guess you could multiply the numeric value by the scale regardless of whether it's "px", "pt", "em", "rem", etc. and get the same effect if you could parse out not only the numeric value but also the units type.

It doesn't seem to me like there's currently any way to set the font weight other than providing a renderText callback, though maybe I'm missing something.

**Comment by TonyGermaneri** [comment]
Drawing text on a canvas is different than drawing text on the DOM.  If you can figure out how to do what you want using this API I'll surely implement it in the grid.

https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Drawing_text

There isn't a global way to change font right now, but since it's created via JS you can write a little script to set all the font values pretty easy:

`Object.keys(grid.style).forEach(function (key) { if (/font$/.test(key)) { grid.style[key] = 'new font'; } });`

It does seem like the font should be settable at the parent level and it should inherit it on all child levels, but since there isn't actually a hierarchy it has to be done manually, that's a bit of work and complexity, might be worth it.

**Comment by domehead100** [comment]
Okay, thanks.  What if either:  a) there is an attribute for the default font, or b) this is pulled off of the canvas element itself (allowing the user to override it with CSS)?  And then, all of the specific font settings default to undefined instead of being given default values.  Then, each time text is drawn, if the specific font is not undefined then that is used, otherwise it sets .font to the default font.

**Comment by TonyGermaneri** [comment]
Yeah, I think a single style property called `font` would work for setting the global font.  I can set the rest of the font values to `inherit` which would be a cue to the drawing function that it should use the `font` property.  Sounds easy to do.

Remember that there is no such thing as CSS here.  Trying to read CSS data is very difficult because of all the potential valid values that CSS can produce along with the ambiguity they imply in their implementation.
